packages/@react-aria/gridlist/src/useGridListItem.ts

```diff
@@ -89,6 +124,18 @@ export function useGridListItem<T>(props: AriaGridListItemOptions, state: ListSt
     let walker = getFocusableTreeWalker(ref.current);
     walker.currentNode = document.activeElement;
 
+    if ('expandedKeys' in state && document.activeElement === ref.current) {
+      if ((e.key === EXPANSION_KEYS['expand'][direction]) && state.selectionManager.focusedKey === node.key && hasChildRows && state.expandedKeys !== 'all' && !state.expandedKeys.has(node.key)) {
+        state.toggleKey(node.key);
+        e.stopPropagation();
+        return;
+      } else if ((e.key === EXPANSION_KEYS['collapse'][direction]) && state.selectionManager.focusedKey === node.key && hasChildRows && (state.expandedKeys === 'all' || state.expandedKeys.has(node.key))) {
+        state.toggleKey(node.key);
+        e.stopPropagation();
+        return;
+      }
+    }
+
```

Comment:
Need to modify the keyboard nav logic here for expansion and stuff since this is a capturing listener and thus can't be stopped in the useTreeGridListItem hook. If we want to keep this hooks clean from tree grid stuff, I can copy pasta the code here into useTreeGridListItem. Alternatively, I could make this hook and useGridList hook keep all the grid specific stuff and get rid of the TreeGridList hooks